source/slang/decl-defs.h

```diff
@@ -90,21 +90,30 @@ SIMPLE_SYNTAX_CLASS(ClassDecl, AggTypeDecl)
 // An interface which other types can conform to
 SIMPLE_SYNTAX_CLASS(InterfaceDecl, AggTypeDecl)
 
+ABSTRACT_SYNTAX_CLASS(TypeConstraintDecl, Decl)
+    RAW(
+    virtual TypeExp& getSup() = 0;
```

Comment:
I'm not a fan of adding `virtual` functions to the AST hierarchy, but I'm also not going to fight it in this case.